datafusion/expr/src/signature.rs

```diff
@@ -346,13 +346,75 @@ impl Signature {
     }
 }
 
-/// Monotonicity of the `ScalarFunctionExpr` with respect to its arguments.
-/// Each element of this vector corresponds to an argument and indicates whether
-/// the function's behavior is monotonic, or non-monotonic/unknown for that argument, namely:
-/// - `None` signifies unknown monotonicity or non-monotonicity.
-/// - `Some(true)` indicates that the function is monotonically increasing w.r.t. the argument in question.
-/// - Some(false) indicates that the function is monotonically decreasing w.r.t. the argument in question.
-pub type FuncMonotonicity = Vec<Option<bool>>;
+#[derive(Debug, Clone, PartialEq)]
```

Comment:
Thanks @tinfoil-knight. I can just say this `PartialEq` derivation could be dangerous here. One can expect to see the equality of Increasing and vec![Some(true)].

Reply:
Yeah. Makes sense. I've fixed this in d47b89664df19ebcd59ad2aae4b8f6d0572f08fd